library/ssl_msg.c

```diff
@@ -2980,15 +2982,15 @@ int mbedtls_ssl_prepare_handshake_record( mbedtls_ssl_context *ssl )
 {
     if( ssl->in_msglen < mbedtls_ssl_hs_hdr_len( ssl ) )
     {
-        MBEDTLS_SSL_DEBUG_MSG( 1, ( "handshake message too short: %d",
+        MBEDTLS_SSL_DEBUG_MSG( 1, ( "handshake message too short: %zu",
                             ssl->in_msglen ) );
         return( MBEDTLS_ERR_SSL_INVALID_RECORD );
     }
 
     ssl->in_hslen = mbedtls_ssl_hs_hdr_len( ssl ) + ssl_get_hs_total_len( ssl );
 
     MBEDTLS_SSL_DEBUG_MSG( 3, ( "handshake message: msglen ="
-                        " %d, type = %d, hslen = %d",
+                        " %zu, type = %hhu, hslen = %zu",
```

Comment:
You can't actually pass a `short`, `char` or `float` to `printf` and friends: they're variadic functions, whose arguments are promoted. Both `%d` and `%u` are correct specifiers. (`%d` is only correct if the actual value is less than `INT_MAX`, which is always the case for Mbed TLS since we don't support platforms where `CHAR_BIT > 8`.) The `h` and `hh` modifiers are only useful for `scanf` and for `%n`. So I think we shouldn't use `%hhu` or even `%hu`, which reduced printf libraries might not implement, and stick to `%u`.

Reply:
Ok, I will admit I wasn't sure myself - %hh was introduced in C99 as well, and proved slightly devisive (some people did not believe it necessary), will reduce to %u in next patch